libass/ass.h

```diff
@@ -734,6 +734,27 @@ void ass_process_chunk(ASS_Track *track, const char *data, int size,
  */
 void ass_set_check_readorder(ASS_Track *track, int check_readorder);
 
+/**
+ * \brief Prune events that preceed deadline if the cache exceeds size_threshold
+ * \param track track
+ * \param deadline cut-off timestamp
+ * \param size_threshold Minimum KiB size to trigger the garbage collection.
+ * Negative values skip the logic.
+ * \return number of events pruned.
+*/
+int ass_prune_history(ASS_Track *track, long long deadline, int size_threshold);
```

Comment:
i think `size_t` and `0` for skipping would be more correct type-wise

In general, does tracking by accumulated event text size bring a real benefit over just number of events?

Reply:
A bitmap/vector event would consume a lot more memory than innocent dialogue. A memory surge could happen if there's a spike of such events within a narrow time window, or the distribution of events type changes vastly as the stream goes. I would say it is more robust and flexible.